### PR TITLE
server: refactor error-handling

### PR DESCRIPTION
**Pull request by lucab**
This refactors http handlers, so that they follow traditional
error handling idiom.

### CODE REVIEW

File: internal/server/pre_reboot.go [hunk 1]
```diff
@@ -18,53 +18,58 @@ const (
...
-		defer cancel()
-		lockManager, err := lock.NewManager(ctx, a.EtcdEndpoints, nodeIdentity.Group, slots)
-		if err != nil {
-			logrus.Errorln("failed to initialize semaphore manager: ", err)
-			http.Error(w, err.Error(), 500)
-			return
-		}
-		defer lockManager.Close()
+// preRebootHandler contains pre-reboot handling logic
+func (a *Airlock) preRebootHandler(w http.ResponseWriter, req *http.Request) (int, error) {
```
Comment (arithx) [inline internal/server/pre_reboot.go]: This function no longer uses the `http.ResponseWriter`, worth dropping it from the parameter list?

File: internal/server/steady_state.go [hunk 2]
```diff
@@ -18,53 +18,59 @@ const (
...
-			logrus.Errorln("failed to initialize semaphore manager: ", err)
-			http.Error(w, err.Error(), 500)
-			return
-		}
-		defer lockManager.Close()
+// steadyStateHandler contains logic to handle steady-state.
+func (a *Airlock) steadyStateHandler(w http.ResponseWriter, req *http.Request) (int, error) {
+	logrus.Debug("got steady-state report")
+	if a == nil {
+		http.Error(w, errNilAirlockServer.Error(), 500)
```
Comment (arithx) [inline internal/server/steady_state.go]: It feels like this should be a `logrus.Errorln` or dropped.

Comment (lucab) [reply internal/server/steady_state.go]: Indeed, good catch, I missed it when moving stuff around!

File: internal/server/steady_state.go [hunk 3]
```diff
@@ -18,53 +18,59 @@ const (
...
 func (a *Airlock) SteadyState() http.Handler {
 	handler := func(w http.ResponseWriter, req *http.Request) {
-		logrus.Debug("got steady-state report")
-		if a == nil {
-			http.Error(w, errNilAirlockServer.Error(), 500)
-			return
-		}
-
-		nodeIdentity, err := validateIdentity(req)
+		code, err := a.preRebootHandler(w, req)
```
Comment (lucab) [inline internal/server/steady_state.go]: Self note: copy-paste error.

File: internal/server/pre_reboot.go (hunk 1 above)
Comment (lucab) [reply internal/server/pre_reboot.go]: I also added an explicit status-code write for the positive case, directly at the calling site.

**Comment by lucab** [comment]
Amended, PTAL.

**Comment by lucab** [comment]
@arithx thanks (once again) for the prompt review!

**Pull request merged**